File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.java
class Solution {
    public int[] solution(int numer1, int denom1, int numer2, int denom2) {
        
        int commonNumer1 = numer1 * denom2;
        int commonNumer2 = numer2 * denom1;
        int commonNumer = commonNumer1+commonNumer2;
        int commonDen = denom1 * denom2; 
        
        int gcd = 0;
        
        for (int i=1; i <= commonNumer && i <= commonDen ; i++){
             if( commonNumer%i == 0 && commonDen%i ==0 ){
                gcd = i;
            }
        }
         commonNumer = commonNumer/gcd;
         commonDen = commonDen/gcd;
        
        
        int[] answer = {commonNumer, commonDen};
        return answer;
    }
}

File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.py
import math 
def solution(numer1, denom1, numer2, denom2):
    answer = []
    son = 0
    mom = 0
    
    
    if math.gcd(denom1, denom2) == 1 :
        son = denom1 * denom2
        mom = (numer1 * denom2) + (numer2 * denom1)
        
        
    else :
        if denom1 < denom2 :
            son = denom1 * math.gcd(denom1, denom2)
            mom = numer1 * math.gcd(denom1, denom2) + numer2
                
            
        elif denom1 > denom2:
            son = denom2 * math.gcd(denom1, denom2)
            mom = numer2 * math.gcd(denom1, denom2)
                
            
        else :
            son = denom1 
            mom = numer1 + numer2
    print(mom, son)
    while (True) :
        if math.gcd(mom, son) != 1 :
            gcd = math.gcd(mom, son)
            mom = mom // gcd
            son = son // gcd
        else :
            break
            
    return mom, son